lib/index.js

```diff
@@ -107,28 +107,40 @@ api.set = async ({
   } else if(type === 'nonce') {
     // FIXME: may need to allow up to N tokens, one per `clientId` if
     // `clientId` is set
-
-    // if there is an existing nonce, delete it, and generate a new one
-    let existingToken;
+    const maxNonceCount = 5;
+    let existingTokens;
     try {
-      // check to see if nonce exists
-      existingToken = await api.get({actor: null, account, email, type});
+      // check if nonce exists
+      existingTokens = await api.get({actor: null, account, email, type});
     } catch(e) {
       if(e.name !== 'NotFoundError') {
         throw e;
       }
     }
-
-    if(existingToken) {
-      try {
-        // the `get` API call did not throw, a nonce exists, delete it
-        await api.remove({actor, account, type});
-      } catch(e) {
-        if(e.name !== 'NotFoundError') {
-          throw e;
+    if(existingTokens) {
+      for(const existingToken of existingTokens) {
+        // check if existing token has expired
+        if(existingToken && new Date() > new Date(existingToken.expires)) {
```

Comment:
We could do `const now = new Date();` before the loop so we don't have to do it each time.